### PR TITLE
Task 6 redux

### PR DESCRIPTION
**Pull request by OlehKhozhai**

### CODE REVIEW

File: src/components/MoviesList/index.tsx [hunk 1]
```diff
@@ -1,34 +1,34 @@
...
-const MoviesList: React.FC<MoviesListProps> = ({ className }) => {
-  const { activeTab } = useSelector(commonSelector);
-
-  const filteredMovies = movies.filter(({ genre }) => genre === activeTab || activeTab === 'all');
-
+const MoviesList: React.FC<MoviesListProps> = ({ movies, isLoading, className }) => {
   return (
     <div className={cn(styles.root, className)}>
-      {filteredMovies.length > 0 && (
+      {movies.length > 0 && (
```
Comment (yurii-hasiuk) [inline src/components/MoviesList/index.tsx:17]: `movies.length && ...` ?

File: src/components/MoviesList/index.tsx [hunk 2]
```diff
@@ -1,34 +1,34 @@
...
         <>
-          <h3 className={styles.counter}>{filteredMovies.length} movies found</h3>
+          <h3 className={styles.counter}>{movies.length} movies found</h3>
 
-          {filteredMovies.map((movie) => {
+          {movies.map((movie) => {
             return <MoviesListItem key={movie.id} {...movie} />;
```
Comment (yurii-hasiuk) [inline src/components/MoviesList/index.tsx:22]: We can omit `return` here

File: src/components/MoviesList/index.tsx [hunk 3]
```diff
@@ -1,34 +1,34 @@
...
           })}
         </>
       )}
 
-      {filteredMovies.length === 0 && <h2 className={styles.noMovieFound}>No movies found</h2>}
+      {movies.length === 0 && !isLoading && (
```
Comment (yurii-hasiuk) [inline src/components/MoviesList/index.tsx:27]: `movies.length && !isLoading && ...` ?

File: src/pages/MovieDetails/index.tsx [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
 
 const MovieDetails = () => {
+  const dispatch = useDispatch();
+  const movieDetails = useSelector(movieDetailsSelector);
+  const { movies, isMoviesLoading } = useSelector(moviesSelector);
+  const params = useParams<{ id: string }>();
+
+  React.useEffect(() => {
+    if (params.id) {
+      dispatch(getMovieDetailsAction(+params.id));
```
Comment (yurii-hasiuk) [inline src/pages/MovieDetails/index.tsx:21]: Can we use destructuring to get `id` from params?

File: src/redux/actions.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+      })
+    );
+  };
+
+export const getMoviesAction =
+  ({ search = '', searchBy = '', sortBy }: GetMoviesAction['payload']) =>
+  (dispatch: Dispatch) => {
+    dispatch({ type: GET_MOVIES });
+
+    fetch(`${BASE_URL}?search=${search}&searchBy=${searchBy}&sortBy=${sortBy}&sortOrder=desc`)
```
Comment (yurii-hasiuk) [inline src/redux/actions.ts:84]: I would split `fetch` and `dispatch` logic. For data fetching we can create a separate helper

File: src/redux/actions.ts [hunk 6]
```diff
@@ -0,0 +1,183 @@
...
+  };
+
+export const getMovieDetailsAction =
+  (id: GetMovieDetailsAction['payload']) => (dispatch: Dispatch) => {
+    dispatch({
+      type: GET_MOVIE_DETAILS,
+      payload: id,
+    } as GetMovieDetailsAction);
+
+    fetch(`${BASE_URL}/${id}`)
```
Comment (yurii-hasiuk) [inline src/redux/actions.ts:107]: Please check all occurrences

File: src/redux/selectors.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import { AppStore } from 'redux/store';
+
+export const activeTabSelector = (store: AppStore) => store.activeTab;
+
+export const moviesSelector = (store: AppStore) => ({
+  movies: store.movies,
+  isMoviesLoading: store.isMoviesLoading,
+});
+
+export const movieDetailsSelector = (store: AppStore) => store.movieDetails;
```
Comment (yurii-hasiuk) [inline src/redux/selectors.ts:10]: Consider usage of  `redux/toolkit`, you can use `createSelector` from there

File: src/redux/types.ts [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+  activeTab: typeof TABS[number];
+  sortBy: keyof typeof DROPDOWN_OPTIONS_DEFINITION;
+  movies: Movie[];
+  isMoviesLoading: boolean;
+  movieDetails: Movie | null;
+  error: string;
+}
+
+// SET_ACTIVE_TAB
+export const SET_ACTIVE_TAB = 'SET_ACTIVE_TAB';
```
Comment (yurii-hasiuk) [inline src/redux/types.ts:28]: I would not mix redux and movie types here.

**Review by yurii-hasiuk — CHANGES_REQUESTED** [review]

File: src/components/MoviesList/index.tsx (hunk 3 above)
Comment (OlehKhozhai) [reply src/components/MoviesList/index.tsx:27]: this condition will return 0 and render it
https://codesandbox.io/s/react-nested-accordion-simple-example-forked-9yprd?file=/src/App.js

File: src/components/MoviesList/index.tsx (hunk 1 above)
Comment (OlehKhozhai) [reply src/components/MoviesList/index.tsx:17]: this condition will return 0 and render it
https://codesandbox.io/s/react-nested-accordion-simple-example-forked-9yprd?file=/src/App.js

File: src/redux/types.ts (hunk 8 above)
Comment (OlehKhozhai) [reply src/redux/types.ts:28]: but it is redux types

File: src/components/MoviesList/index.tsx (hunk 2 above)
Comment (OlehKhozhai) [reply src/components/MoviesList/index.tsx:22]: yep, but it not so necessary

File: src/pages/MovieDetails/index.tsx (hunk 4 above)
Comment (OlehKhozhai) [reply src/pages/MovieDetails/index.tsx:21]: yep

**Pull request merged**